bokeh/io.py

```diff
@@ -439,17 +434,13 @@ def _get_save_args(state, filename, resources, title):
 
     return filename, resources, title
 
-def _save_helper(obj, filename, resources, title, validate):
+def _save_helper(obj, state, filename, resources, title, validate):
     remove_after = False
-    if isinstance(obj, LayoutDOM):
-        if obj.document is None:
-            Document().add_root(obj)
```

Comment:
See what was done for themes in https://github.com/bokeh/bokeh/pull/5932/files I think we have to do the same thing. Is there a reason `save` can't just be implemented in terms of `components` ?